projects/angular/src/data/datagrid/_datagrid.clarity.scss

```
@@ -891,11 +904,16 @@
       }
     }
 
-    .datagrid-detail-caret.datagrid-cell {
-      padding: 0;
+    .datagrid-detail-caret {
+      &.datagrid-cell {
+        padding: 0;
+      }
+      .datagrid-detail-caret-button {
+        @include datagrid-action-button-styles(tables-variables.$clr-table-compact-row-height);
+      }
     }
 
-    .datagrid-cell {
+    .datagrid-cell:not(.datagrid-detail-caret):not(.datagrid-expandable-caret):not(.datagrid-row-actions) {
```

You ca ndo it in a single `not`

```suggestion
    .datagrid-cell:not(.datagrid-detail-caret, .datagrid-expandable-caret, .datagrid-row-actions) {
```

---

Fixed in [6c3bdf4](https://github.com/vmware-clarity/ng-clarity/pull/1818/commits/6c3bdf41885e2d8a6fb38c34b6d35312c1c3a2d9)